src/rgw/rgw_rest_swift.cc

```diff
@@ -2805,7 +2805,10 @@ int RGWHandler_REST_SWIFT::postauth_init(optional_yield y)
   struct req_init_state* t = &s->init_state;
 
   /* XXX Stub this until Swift Auth sets account into URL. */
-  s->bucket_tenant = s->user->get_tenant();
+  if(g_conf()->rgw_swift_account_in_url)
+    s->bucket_tenant = s->account_name;
```

Comment:
The only way I see the segfault possible from this change is if `account_name` is not pointing at something that exists?

It seems that there are cases where it can be empty looking at this code base:

https://github.com/ceph/ceph/blob/9a911f0cb5f7f4511274cf44d23b74c96f1a3863/src/rgw/rgw_rest_swift.cc#L2088

Does it make sense to also add that check as well, so make it something like this?

```suggestion
  if (g_conf()->rgw_swift_account_in_url && !s->account_name.empty())
    s->bucket_tenant = s->account_name;
```

Reply:
i'll retest with this extra check

Reply:
same crash with that change, http://qa-proxy.ceph.com/teuthology/cbodley-2022-01-17_21:32:56-rgw:tempest-wip-rgw-swift-anonymous-tenant-distro-default-smithi/6622675/teuthology.log